timer reschedule error message

I inadvertently called `timer.reschedule an_int_variable` instead of `timer.reschedule `an_int_variable.seconds` and this happened:

```
org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `to_ms' for 50:Integer
Did you mean?  to_s
               to_c
               to_r
               to_f
               to_i
	at RUBY.reschedule(/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-2.9.2/lib/openhab/core/dsl/timers.rb:67) ~[?:?]
	at RUBY.<main>(<script>:149) ~[?:?]
	at RUBY.initialize(/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-2.9.2/lib/openhab/core/dsl/timers.rb:46) ~[?:?]
	at org.jruby.ext.thread.Mutex.synchronize(org/jruby/ext/thread/Mutex.java:164) ~[?:?]
	at RUBY.initialize(/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-2.9.2/lib/openhab/core/dsl/timers.rb:45) ~[?:?]
```


Should we take bare integer argument to reschedule as seconds? I know it's a bit ambiguous though. Or at the least, issue a better error message?

How about this:

```
07:30:50.378 [ERROR] [ript.internal.ScriptEngineManagerImpl] - Error during evaluation of script 'file:/openhab/conf/automation/jsr223/ruby/personal/test.rb': org.jruby.embed.EvalFailedException: (ArgumentError) Invalid duration argument. It does not respond to to_ms
```

With the new Java Duration, the error message is now:
```
Caused by: org.jruby.exceptions.NameError: (NameError) no method 'plus' for arguments (org.jruby.RubyFixnum) on Java::JavaTime::ZonedDateTime
  available overloads:
    (java.time.temporal.TemporalAmount)
    (java.time.temporal.TemporalAmount)
    (java.time.temporal.TemporalAmount)
```

To improve this error message, we'd need to do checks in a few places: 
* after
* changed duration
* cron

Is it worth doing it or leave it as is?

--- a/lib/openhab/dsl/timers/timer.rb
+++ b/lib/openhab/dsl/timers/timer.rb
@@ -59,6 +59,10 @@ module OpenHAB
         # @return [Timer] Rescheduled timer instances
         #
         def reschedule(duration = nil)
+          unless duration.nil? || duration.is_a?(Java::JavaTimeTemporal::TemporalAmount)
+            raise ArgumentError, 'Supplied argument must be a duration'
+          end
+
           duration ||= @duration
           Timers.timer_manager.add(self)
           @timer.reschedule(ZonedDateTime.now.plus(duration))

